pgjdbc/src/main/java/org/postgresql/jdbc/TimestampUtils.java

```diff
@@ -809,6 +847,61 @@ public Timestamp toTimestampBin(TimeZone tz, byte[] bytes, boolean timestamptz)
     return ts;
   }
 
+  //#if mvn.project.property.postgresql.jdbc.spec >= "JDBC4.2"
+  /**
+   * Returns the SQL Timestamp object matching the given bytes with {@link Oid#TIMESTAMP} or
+   * {@link Oid#TIMESTAMPTZ}.
+   *
+   * @param bytes The binary encoded timestamp value.
+   * @return The parsed timestamp object.
+   * @throws PSQLException If binary format could not be parsed.
+   */
+  public LocalDateTime toLocalDateTimeBin(byte[] bytes)  throws PSQLException {
```

Comment:
You're correct (about the copy and paste), I'll refactor.

Now for why it's needed: The Timestamp semantics are a local date time in the JVM default time zone. But what if that value "doesn't exist" because of a DST offset transition? Take for example the value '2016-03-27 02:15:00'. This does not exist in Europe/Berlin because at '2016-03-27 02:00:00' the clocks are moved to '2016-03-27 03:00:00'. So what happens to the value we read from the database? We get '2016-03-27 03:15:00' instead of '2016-03-27 02:15:00'. The only way to avoid this is not to covert to Timestamp first but instead compute the value directly.